test/battle/CHealthTest.cpp

```diff
@@ -0,0 +1,221 @@
+}
+
+BOOST_AUTO_TEST_CASE(damage)
+{
+	CUnitHealthInfoMock uhi;
+	CHealth health(&uhi);
+	health.init();
+
+	checkNormalDamage(health, 0);
+	checkFullHealth(health, uhi);
+
+	checkNormalDamage(health, uhi.maxHealth - 1);
+	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT);
+	BOOST_CHECK_EQUAL(health.getFirstHPleft(), 1);
+	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+
+	checkNormalDamage(health, 1);
+	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT - 1);
+	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH);
+	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+
+	checkNormalDamage(health, UNIT_HEALTH * (UNIT_AMOUNT - 1));
+	checkEmptyHealth(health, uhi);
+
+	checkNoDamage(health, 1337);
+	checkEmptyHealth(health, uhi);
+}
+
+BOOST_AUTO_TEST_CASE(heal)
+{
+	CUnitHealthInfoMock uhi;
+	CHealth health(&uhi);
+	health.init();
+	checkNormalDamage(health, 99);
+	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT);
+	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH-99);
+	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+
+	checkHeal(health, EHealLevel::HEAL, EHealPower::PERMANENT, 9, 9);
+	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT);
+	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH-90);
+	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+
+	checkHeal(health, EHealLevel::RESURRECT, EHealPower::ONE_BATTLE, 40, 40);
+	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT);
+	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH-50);
+	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+
+	checkHeal(health, EHealLevel::OVERHEAL, EHealPower::PERMANENT, 50, 50);
+	checkFullHealth(health, uhi);
+}
+
+BOOST_AUTO_TEST_CASE(resurrectOneBattle)
+{
+	CUnitHealthInfoMock uhi;
+	CHealth health(&uhi);
+	health.init();
+
+	checkNormalDamage(health, UNIT_HEALTH);
+	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT - 1);
+	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH);
+	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+
+	checkHeal(health, EHealLevel::RESURRECT, EHealPower::ONE_BATTLE, UNIT_HEALTH, UNIT_HEALTH);
+	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT);
+	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH);
+	BOOST_CHECK_EQUAL(health.getResurrected(), 1);
+
+	checkNormalDamage(health, UNIT_HEALTH);
+	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT - 1);
+	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH);
+	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+
+	health.init();
+
+	checkNormalDamage(health, UNIT_HEALTH * UNIT_AMOUNT);
+	checkEmptyHealth(health, uhi);
+
+	checkHeal(health, EHealLevel::RESURRECT, EHealPower::ONE_BATTLE, UNIT_HEALTH * UNIT_AMOUNT, UNIT_HEALTH * UNIT_AMOUNT);
+	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT);
+	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH);
+	BOOST_CHECK_EQUAL(health.getResurrected(), UNIT_AMOUNT);
+
+	health.takeResurrected();
+	checkEmptyHealth(health, uhi);
+}
+
+BOOST_AUTO_TEST_CASE(resurrectPermanent)
+{
+	CUnitHealthInfoMock uhi;
+	CHealth health(&uhi);
+	health.init();
+
```

Comment:
For me this is code duplication. You should make fixture for handling it. You can even change CUnitHealthInfoMock class for that.